runtime/runtime_events.c

```diff
@@ -573,3 +602,154 @@ void caml_ev_alloc_flush() {
     alloc_buckets[i] = 0;
   }
 }
+
+/* Registers the [index] -> [event_name] mapping in the dedicated space in the
+   ring buffer */
+void events_register_write_buffer(int idx, value event_name) {
+  struct runtime_events_custom_event *custom_event =
+    &((struct runtime_events_custom_event *)
+      ((char *)current_metadata + current_metadata->custom_events_offset))[idx];
+
+  // TODO: what if caml string is not C safe ?
+  int length = caml_string_length(event_name);
+  strncpy(custom_event->name, String_val(event_name), length);
+  custom_event->name[length] = 0; // necessary ? not sure.
+}
+
+CAMLprim value caml_runtime_events_user_register(value event_name,
+  value event_tag, value event_type)
+{
+  CAMLparam1(event_name);
```

Comment:
Don't all of the parameters above need to be registered here? `CAMLparam3(event_name, event_tag, event_type)l`

Reply:
Thanks for taking a look, yes indeed ! 
Overall I need to perform an additional pass on the code to make sure that:
- C / OCaml ffi rules are respected
- code is domain-safe